.github/workflows/build.yaml

```diff
@@ -96,7 +108,7 @@ jobs:
         if: ${{ runner.os == 'Linux' && !failure() }}
         uses: actions/upload-artifact@v4.3.4
         with:
-          name: liberty-tools-intellij-LTI-main-LSP4IJ-${{ env.LSP4IJ_BRANCH }}
+          name: liberty-tools-intellij-LTI-${{ env.REF_LTI_TAG }}-LSP4IJ-${{ env.LSP4IJ_BRANCH }}
```

Comment:
As part of resolving the merge conflict, I found that this change is required. With this change, how will the artifact name look as shown in the screenshot below?
Cron Job build: Link : https://github.com/vaisakhkannan/liberty-tools-intellij/actions/runs/9988499251
<img width="838" alt="Screenshot 2024-07-18 at 2 02 21 PM" src="https://github.com/user-attachments/assets/2c3d444e-ea95-487c-b991-3514e835b59d">

Normal Build:  Link : https://github.com/vaisakhkannan/libe
<img width="1040" alt="Screenshot 2024-07-18 at 2 43 41 PM" src="https://github.com/user-attachments/assets/807bdc05-96c3-40ff-b0f1-56478db2c00d">
rty-tools-intellij/actions/runs/9988666800

The main point is, if this value `"${{ env.REF_LTI_TAG }}"` is empty ( see the image above), it means there is no value present. We can't simply put a default value like this: `LSP4IJ_BRANCH: ${{ inputs.lsp4ijBranch || 'latest' }}`, because `lsp4ijBranch `doesn’t require a default value (shown in the image below) in order to provide the 'latest' value.

<img width="675" alt="Screenshot 2024-07-18 at 2 49 06 PM" src="https://github.com/user-attachments/assets/69b905bd-2c6e-4ece-a2bc-96148859a24a">

Similarly, we can't provide a similar value because the `refLTITag` parameter requires a default value of 'main'.

Reply:
A few questions/comments:

1. What does `latest` mean for the LSP4IJ branch? Does that mean using the `main` branch? If so, I think it would be better to use `main` in the artifact name.
2. It looks like the default value for `refLTITag` is `main`. How does this variable end up being empty?
3. If `refLTITag` is empty, how does the checkout part of the workflow work? https://github.com/OpenLiberty/liberty-tools-intellij/blob/76f216e3d88006553fdc9b9909c7d36dffe08f16/.github/workflows/build.yaml#L77-L81
4. If it is acceptable for `refLTITag` to be empty, is there a way to check if it is empty, and if so, assign it a value of `main` before using it in the artifact name or other places in the workflow?

Reply:
@TrevCraw ,
1.` latest` doesn't mean that it is using the `main` branch.`latest` points to `lsp4ijVersion` https://github.com/OpenLiberty/liberty-tools-intellij/blob/b34b7af4d6db48c74c4d0bd5b92531919081dc73/build.gradle#L15.For normal builds --`USE_LOCAL_PLUGIN: false` Anusree made this change to get the value in normal builds (push/PR --> will print 'latest' value). She simply put 'latest' [here](https://github.com/OpenLiberty/liberty-tools-intellij/blob/b34b7af4d6db48c74c4d0bd5b92531919081dc73/.github/workflows/build.yaml#L57) because there is no default value specified in the [workflow_call](https://github.com/OpenLiberty/liberty-tools-intellij/blob/b34b7af4d6db48c74c4d0bd5b92531919081dc73/.github/workflows/build.yaml#L14).
2.

Reply:
Thank you for the updates @vaisakhkannan. Can you change the name to be "default" for LSP4IJ as well? "latest" will not always be accurate, since it will be whatever is defined in the `build.gradle`.

Reply:
Yes, Thank you @TrevCraw , I did the change.